source/particle/world.cc

```diff
@@ -979,25 +979,37 @@ namespace aspect
 
       std::vector<Tensor<1,dim> >  result(particles_in_cell);
       std::vector<Tensor<1,dim> >  old_result(particles_in_cell);
-      std::vector<Point<dim> >     particle_points(particles_in_cell);
 
-      typename std::multimap<types::LevelInd, Particle<dim> >::iterator it = begin_particle;
-      for (unsigned int i = 0; it!=end_particle; ++it,++i)
+      std::vector<types::global_dof_index> cell_dof_indices (this->get_fe().dofs_per_cell);
+
+      const FiniteElement<dim> &velocity_fe = this->get_fe().base_element(this->introspection().base_elements.velocities);
+      cell->get_dof_indices (cell_dof_indices);
+
+      Table<2,double> shape_values (velocity_fe.dofs_per_cell,particles_in_cell);
+
+      for (unsigned int j=0; j<velocity_fe.dofs_per_cell; ++j)
         {
-          particle_points[i] = it->second.get_reference_location();
+          typename std::multimap<types::LevelInd, Particle<dim> >::iterator it = begin_particle;
+          for (unsigned int i = 0; it!=end_particle; ++it,++i)
+            shape_values(j,i) = velocity_fe.shape_value(j,it->second.get_reference_location());
         }
 
-      const Quadrature<dim> quadrature_formula(particle_points);
-      FEValues<dim> fe_value (this->get_mapping(),
-                              this->get_fe(),
-                              quadrature_formula,
-                              update_values);
+      for (unsigned int j=0; j<velocity_fe.dofs_per_cell; ++j)
```

Comment:
I think you can make the code more local (and allocate less memory) if you merge this loop with the previous one (which also goes over `j`). You then won't need to fill a whole `Table<2,double>`, but could get away with just a `std::vector<double>` of length `particles_in_cell` that you re-use for all `j`.